src/transformers/modeling_utils.py

```diff
@@ -4347,6 +4395,13 @@ def from_pretrained(
             if not is_fsdp_enabled() and not is_deepspeed_zero3_enabled():
                 dispatch_model(model, **device_map_kwargs)
 
+        # This is needed for the RotaryEmbedding, which was not initialized on the correct device as it is
+        # not part of the state_dict (persistent=False)
+        if device_mesh is not None:
+            for buffer in model.buffers():
+                if buffer.device != tp_device:
+                    buffer.data = buffer.to(tp_device)
```

Comment:
Interesting, remember that no we pass the cos and sin as input to all layers so to are passed